cli/src/lockfiles/mod.rs

```diff
@@ -20,12 +19,15 @@ pub use ruby::GemLock;
 
 pub type ParseResult = anyhow::Result<Vec<PackageDescriptor>>;
 
-pub trait Parseable {
-    fn new(filename: &Path) -> Result<Self, io::Error>
-    where
-        Self: Sized;
-
-    fn parse(&self) -> ParseResult;
+pub trait Parse {
+    /// Parse from a string
+    fn parse(&self, data: &str) -> ParseResult;
+    /// Indicate the type of packages parsed by this parser
+    fn package_type(&self) -> PackageType;
+}
 
-    fn package_type() -> PackageType;
+/// Parse a file with the given parser.
+pub fn parse_file<T: Parse, P: AsRef<Path>>(parser: T, path: P) -> ParseResult {
+    let data = read_to_string(path)?;
+    parser.parse(&data)
 }
```

Comment:
This might as well be a method on the trait, right? Like one that is already implemented by default.

Reply:
Yep. I think that makes sense